pytensor/tensor/type.py

```diff
@@ -794,7 +817,11 @@ def tensor(*args, **kwargs):
 ulscalar = TensorType("uint64", ())
 
 
-def scalar(name=None, dtype=None):
+def scalar(
+    name: Optional[str] = None,
+    *,
+    dtype: Optional["DTypeLike"] = None,
+) -> "TensorVariable":
```

Comment:
Is this the public API `pt.scalar()` ?

Why the breaking change?

Reply:
You mean forcing `dtype` to be kwarg only?